x-pack/functionbeat/provider/aws/cli_manager.go

```diff
@@ -106,29 +106,37 @@ func (c *CLIManager) template(function installer, name, codeLoc string) *cloudfo
 	// Merge any specific policies from the service.
 	policies = append(policies, function.Policies()...)
 
-	// Create the roles for the lambda.
 	template := cloudformation.NewTemplate()
-	// doc: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html
-	template.Resources[prefix("")+"IAMRoleLambdaExecution"] = &cloudformation.AWSIAMRole{
-		AssumeRolePolicyDocument: map[string]interface{}{
-			"Statement": []interface{}{
-				map[string]interface{}{
-					"Action": "sts:AssumeRole",
-					"Effect": "Allow",
-					"Principal": map[string]interface{}{
-						"Service": cloudformation.Join("", []string{
-							"lambda.",
-							cloudformation.Ref("AWS::URLSuffix"),
-						}),
+
+	role := lambdaConfig.Role
+	dependsOn := make([]string, 0)
+	if lambdaConfig.Role == "" {
```

Comment:
I think we should log that we are using a custom role, I presume that we will get a few questions concerning the policies required if they use their role. So I think we will need to create a followup issue doc issue to describe them. Can you coordinate with @dedemorton for that?

Reply:
Sure.

Reply:
Created issue with the required policies: https://github.com/elastic/beats/issues/11787